Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h

```diff
@@ -186,6 +184,289 @@ degenerate_faces(const TriangleMesh& tm, OutputIterator out)
+/// @tparam NamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+///
+/// @param f the face to check whether is degenerate
+/// @param tm the triangulated surface mesh upon evaluation
+/// @param np optional \ref pmp_namedparameters "Named Parameters" described below
+///
+/// \cgalNamedParamsBegin
+///    \cgalParamBegin{vertex_point_map} the property map with the points associated to the vertices of `pmesh`. The type of this map is model of `ReadWritePropertyMap`.
+/// If this parameter is omitted, an internal property map for
+/// `CGAL::vertex_point_t` should be available in `PolygonMesh`
+/// \cgalParamEnd
+///    \cgalParamBegin{geom_traits} a geometric traits class instance.
+///       The traits class must provide the nested type `Point_3`,
+///       and the nested functor :
+///         - `Collinear_3` to check whether 3 points are collinear
+///   \cgalParamEnd
+/// \cgalNamedParamsEnd
+///
+/// \return true if the triangle face is degenerate
+template <typename TriangleMesh, typename NamedParameters>
+bool is_degenerate_triangle_face(typename boost::graph_traits<TriangleMesh>::face_descriptor f,
+                                 const TriangleMesh& tm,
+                                 const NamedParameters& np)
+{
+  CGAL_assertion(CGAL::is_triangle_mesh(tm));
+
+  using boost::get_param;
+  using boost::choose_param;
+
+  typedef typename GetVertexPointMap<TriangleMesh, NamedParameters>::const_type VertexPointMap;
+  VertexPointMap vpmap = choose_param(get_param(np, internal_np::vertex_point),
+                                      get_const_property_map(vertex_point, tm));
+  typedef typename GetGeomTraits<TriangleMesh, NamedParameters>::type Traits;
+  Traits traits = choose_param(get_param(np, internal_np::geom_traits), Traits());
+
+  // call from BGL helpers
+  return is_degenerate_triangle_face(f, tm, vpmap, traits);
+}
+
+template <typename TriangleMesh>
+bool is_degenerate_triangle_face(typename boost::graph_traits<TriangleMesh>::face_descriptor f,
+                                 const TriangleMesh& tm)
+{
+  return CGAL::Polygon_mesh_processing::is_degenerate_triangle_face(f, tm, parameters::all_default());
+}
+
+/// \ingroup PMP_repairing_grp
+/// checks whether a triangle face is needle-like.
+/// In a needle-like triangle its longest edge is much longer than the shortest one.
+///
+/// @tparam TriangleMesh a model of `FaceListGraph` and `MutableFaceGraph`
+/// @tparam NamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+///
+/// @param f the face to check whether is almost degenerate
+/// @param tm the triangulated surface mesh upon evaluation
+/// @param threshold a number in the range [0, 1] to indicate the tolerance
+///        upon which to characterize the degeneracy. 1 means that needle triangles
+///        are those that have a infinitely small edge, while 0 means that needle triangles
+///        are those that would have an infinitely long edge
```

Comment:
I don't understand what you mean here. Wouldn't it be easier to use the ratio between the larger and the smallest edge?